eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/ZmqWatcher.scala

```diff
@@ -80,12 +79,20 @@ class ZmqWatcher(blockCount: AtomicLong, client: ExtendedBitcoinClient)(implicit
       context become watching(watches, watchedUtxos, block2tx, Some(task))
 
     case TickNewBlock =>
-      client.getBlockCount.map {
-        case count =>
+      val blockHeigthF = client.getBlockCount.map { count =>
           log.debug(s"setting blockCount=$count")
           blockCount.set(count)
           context.system.eventStream.publish(CurrentBlockCount(count))
+          count
       }
+
+      // if there are pending WatchConfirmed(s) we trigger a rescan to be able to check if any of them was confirmed
+      // this operation is blocking but we only rescan from the last -1 block which should be quick
+      for {
+        blockHeigth <- blockHeigthF
+        _ <- if(watches.exists(w => w.isInstanceOf[WatchConfirmed])) client.rescanBlockChain(blockHeigth - 1) else Future.successful(Unit)
+      } yield ()
+
```

Comment:
I don't understand how limiting rescans to the last block works. Is there a test for this ? What happends when the tx was confirmed while eclair was offline. At least there needs to be a more explicit comment in the code and the commit description, and a specific test

Reply:
+1 for the test, i will add it. If eclair was offline it will re-add the watchers when restarted, i see the point that if it was offline for too long and the transaction went buried in several blocks this won't work. But how should we handle this? If the watcher is added and eclair goes offline how deep should it rescan when restarted? Or how deep should it rescan when the watcher is re-added? I struggle to see how we can get this right without scanning from the beginning every time.